### PR TITLE
[SAGE-540] add Video Thumbnail component 

### PR DESCRIPTION
**Pull request by QuintonJason**
This is a new PR created after the removal of the themes folder in this PR: https://github.com/Kajabi/sage-lib/pull/1497

## Description

- [x] add video thumbnail component (Rails)
- [x] add video thumbnail component (React)
- [x] add trigger to both
- [ ] ~add Video Thumbnail to hero~ -> [SAGE-726](https://kajabi.atlassian.net/browse/SAGE-726)

Design Spec: [KP Payouts -> Video Modal](https://www.figma.com/file/CEfffcNPgGDcDmknNt0HPj/COMM-1616-KP-Payouts?node-id=2822%3A22627)

## Screenshots

|  Before  |  After  |
|--------|--------|
|videoThumbnailRails|videoThumbnailReact|

## Testing in `sage-lib`

Visit the Video Thumbnail views and verify that it matches the design:
- [Rails](http://localhost:4000/pages/component/video_thumbnail?tab=preview#)
- [React](http://localhost:4110/?path=/story/sage-videothumbnail--default)

## Testing in `kajabi-products`

1. (**LOW**) Adds new Video Thumbnail component.

## Related

Closes [SAGE-540](https://kajabi.atlassian.net/browse/SAGE-540)

### CODE REVIEW

**Comment by QuintonJason** [comment]
Bringing over this comment from original PR:
https://github.com/Kajabi/sage-lib/pull/1497#discussion_r919231891

File: docs/app/views/examples/components/video_thumbnail/_rules_dont.html.erb [hunk 1]
```diff
@@ -0,0 +1,49 @@
+<h3>Default Video Thumbnails</h3>
```
Comment (monkeypox8) [inline docs/app/views/examples/components/video_thumbnail/_rules_dont.html.erb:1]: Was this left in by accident? Looks to be a duplicate of the preview markup

File: docs/app/views/examples/components/video_thumbnail/_preview.html.erb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        href: "#",
+      },
+      image_url: "/assets/card-placeholder-hero-24ea70f1ef1e9da27f8f2901667eb03ac7aaab5b82e870c8808835e432ddd3ce.jpg"
+    } %>
+  <% end %>
+  <p>By default, Panel figures are rendered using the image's native aspect ratio and scale accordingly.</p>
+<% end %>
+
+<%# Video player modal %>
+<%= sage_component SageModal, { id: "cool-modal" } do %>
```
Comment (monkeypox8) [inline docs/app/views/examples/components/video_thumbnail/_preview.html.erb:16]: :sunglasses:

File: docs/lib/sage_rails/app/views/sage_components/_sage_video_thumbnail.html.erb [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<%# Desired HTML %>
+<div class="
+    sage-video-thumbnail
+    <%= component.generated_css_classes %>
+  "
+  style="background-image: url(<%= component.image_url %>);"
```
Comment (monkeypox8) [inline docs/lib/sage_rails/app/views/sage_components/_sage_video_thumbnail.html.erb:6]: (non-blocking, future follow-up item) given that we won't be able to enforce appropriate image sizes and compression when in use, it'd seem beneficial for us to move away from background images. If we use `img` elements across the board, we can take advantage of the various built-in browser performance benefits.

File: packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  background-color: transparent;
+
+  &::before {
+    @include sage-icon-base(play);
+    position: absolute;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    z-index: sage-z-index(default, 1);
+    width: rem(24px);
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss:30]: (non-blocking) Thoughts on setting these sizes (and those of the background) as variables for future theming adjustments? The play button does appear fairly large at mobile breakpoints vs the thumbnail, so I could see us wanting to tweak that later.

File: packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  position: relative;
+  width: 100%;
+  height: 100%;
+  aspect-ratio: 16 / 9;
+  background-position: center center;
+  background-repeat: no-repeat;
+  background-size: cover;
+
+  &:focus-within {
+    @include sage-focus-ring;
```
Comment (monkeypox8) [inline packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss:12]: I'm unable to see a focus state on the thumbnail or button. Looks to be intentional, judging by line 55. If that's the case, I'd still recommend we display some sort of visual indication that the modal trigger is focused.

**Review by monkeypox8 — CHANGES_REQUESTED** [review]
This is looking great. One minor cleanup item on the "Don'ts" bit and a question on the focus state to consider.

File: packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss (hunk 5 above)
Comment (QuintonJason) [reply packages/sage-assets/lib/stylesheets/components/_video_thumbnail.scss:12]: Great callout, this has to do with the `overflow: hidden;` in the parent `sage-panel__figure`. I'll need to add a followup Jira ticket to address.

**Comment by QuintonJason** [comment]
Closed as item is no longer needed. See ticket [SAGE-540](https://kajabi.atlassian.net/browse/SAGE-540)

**Pull request closed without merging**